packages/core/src/plugins/meta/itunesmusic.ts

```diff
@@ -0,0 +1,152 @@
+import _ from 'lodash';
+
+import MetaProvider from '../metaProvider';
+import * as iTunes from '../../rest/iTunes';
+import { 
+  SearchResultsArtist, 
+  SearchResultsAlbum, 
+  SearchResultsTrack, 
+  ArtistDetails, 
+  AlbumDetails,
+  SearchResultsSource, 
+  AlbumType
+} from '../plugins.types';
+import LastFmApi from '../../rest/Lastfm';
+import Track from '../../structs/Track';
+import { LastFmArtistInfo, LastfmTopTracks } from '../../rest/Lastfm.types';
+
+class iTunesMusicMetaProvider extends MetaProvider {
+  lastfm: LastFmApi;
+  constructor() {
+    super();
+    this.name = 'iTunesMusic Meta Provider';
+    this.sourceName = 'iTunesMusic Meta Provider';
+    this.description = 'Metadata provider that uses iTunes as a source.';
+    this.searchName = 'iTunesMusic';
+    this.image = null;
+    this.lastfm = new LastFmApi(process.env.LAST_FM_API_KEY, process.env.LASTFM_API_SECRET);
+  }
+
+  async searchForArtists(query: string): Promise<SearchResultsArtist[]> {
+    const artistInfo = (await ((await iTunes.artistSearch(query, '50')).json()));
+    return artistInfo.results
+      .filter(artist => artist.artistType === 'Artist')
+      .map(artist => ({
+        id: artist.artistId,
+        coverImage: '',
```

Comment:
Too bad they don't have images for artists 😔

Reply:
Yeah, I tried but it would require a lot of API request :/